### ISSUE TITLE
When using json logformat, not all logoutput is json

### ISSUE THREAD

**Issue by ArgTang** [argoproj/argo-cd#5641]
If you are trying to resolve an environment-specific issue or have a one-off question about the edge case that does not require a feature then please consider asking a question in argocd slack [channel](https://argoproj.github.io/community/join-slack).

Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

When setting logformat to json not all lines are printed with json:

##### argocd server 
```json
{"grpc.code":"OK","grpc.method":"WatchResourceTree","grpc.service":"application.ApplicationService","grpc.start_time":"2021-03-01T13:56:00Z","grpc.time_ms":3956.718,"level":"info","msg":"finished streaming call with code OK","span.kind":"server","system":"grpc","time":"2021-03-01T13:56:04Z"}
Failed to obtain reader, failed to marshal fields to JSON, json: error calling MarshalJSON for type *grpc.jsonpbMarshalleble: invalid character '\x13' looking for beginning of value
Failed to obtain reader, failed to marshal fields to JSON, json: error calling MarshalJSON for type *grpc.jsonpbMarshalleble: invalid character '\x13' looking for beginning of value
{"grpc.code":"OK","grpc.method":"Watch","grpc.service":"application.ApplicationService","grpc.start_time":"2021-03-01T13:56:09Z","grpc.time_ms":4845.675,"level":"info","msg":"finished streaming call with code OK","span.kind":"server","system":"grpc","time":"2021-03-01T13:56:14Z"}
{"grpc.code":"OK","grpc.method":"WatchResourceTree","grpc.service":"application.ApplicationService","grpc.start_time":"2021-03-01T13:56:09Z","grpc.time_ms":4845.878,"level":"info","msg":"finished streaming call with code OK","span.kind":"server","system":"grpc","time":"2021-03-01T13:56:14Z"}
Failed to obtain reader, failed to marshal fields to JSON, json: error calling MarshalJSON for type *grpc.jsonpbMarshalleble: invalid character '\x13' looking for beginning of value
Failed to obtain reader, failed to marshal fields to JSON, json: error calling MarshalJSON for type *grpc.jsonpbMarshalleble: invalid character '\x13' looking for beginning of value
Failed to obtain reader, failed to marshal fields to JSON, json: error calling MarshalJSON for type *grpc.jsonpbMarshalleble: invalid character '\x00' in string literal
Failed to obtain reader, failed to marshal fields to JSON, json: error calling MarshalJSON for type *grpc.jsonpbMarshalleble: unexpected end of JSON input
```

##### argocd repo-server
```
{"dir": "/tmp/https:__myteam@dev.azure.com_myteam_myrepo__git_myrepo","execID": "lpAfM","level": "info","msg": "git submodule update --init --recursive","time": "2021-03-01T14:06:31Z"}
time="2021-03-01T14:06:31Z" level=info msg=Trace args="[git submodule update --init --recursive]" dir="/tmp/https:__myteam@dev.azure.com_myteam_myrepo__git_myrepo" operation_name="exec git" time_ms=63.853768
{"dir": "/tmp/https:__myteam@dev.azure.com_myteam_myrepo__git_myrepo","execID": "ugV4l","level": "info","msg": "git clean -fdx","time": "2021-03-01T14:06:31Z"}
time="2021-03-01T14:06:31Z" level=info msg=Trace args="[git clean -fdx]" dir="/tmp/https:__myteam@dev.azure.com_myteam_myrepo__git_myrepo" operation_name="exec git" time_ms=4.653347
{"dir": "/tmp/https:__myteam@dev.azure.com_myteam_myrepo__git_myrepo","execID": "OOip1","level": "info","msg": "git rev-parse HEAD","time": "2021-03-01T14:06:31Z"}
time="2021-03-01T14:06:31Z" level=info msg=Trace args="[git rev-parse HEAD]" dir="/tmp/https:__myteam@dev.azure.com_myteam_myrepo__git_myrepo" operation_name="exec git" time_ms=1.640422
```

**To Reproduce**
Using helm add json as logoutput: 
```
argo-cd:
  controller:
    logFormat: json
  server:
    logFormat: json
  repoServer:
    logFormat: json
```
or add  `--logformat json` flag to deployments

**Expected behavior**

If jsonlogging is turned on  i expect all logoutput to be in json format.

**Version**

```shell
argocd@argocd-repo-server-5d7596d999-8gpdp:~$ argocd version
argocd: v1.8.3+0f9c684
  BuildDate: 2021-01-21T22:19:20Z
  GitCommit: 0f9c68427882bf4633d395cbfcd7c9271795fd9b
  GitTreeState: clean
  GoVersion: go1.14.12
  Compiler: gc
  Platform: linux/amd64
```

**Comment by alexmt** [comment]
I believe this is fixed by https://github.com/argoproj/argo-cd/pull/5446 . @shokada can you please double-check me? Your PR fixes the same issue, right ?

**Comment by shokada** [comment]
@alexmt 
Yeah, `Failed to obtain reader... ` error should have been fixed by #5446.

**Comment by alexmt** [comment]
thank you for confirming!
